app/views/forms/view/chm/national-report-6.vue

```diff
@@ -0,0 +1,376 @@
+
+        
+            <section v-if="document.relevantInformation || document.relevantDocuments">
+                <legend>{{ t("additionalInformation") }}</legend>                
+                <view-relevant-information :relevant-information="document.relevantInformation" :relevant-documents="document.relevantDocuments" :locale="locale"> 
+                </view-relevant-information> 
+            </section>           
+        
+            <div> 
+                <ng v-vue-ng:view-record-reference  v-model:ng-model="document.header.identifier" :locales="locale" html></ng>                        
+            </div>         
+        </div>  
+        <!-- TODO: add footer  -->
+        <!-- <ng v-vue-ng:document-metadata  :document="document"></ng>  -->
+        
+    </div>
+</template>
+<script setup>
+    import { computed, onMounted,ref} from 'vue'; 
+    import '~/components/scbd-angularjs-controls/form-control-directives/km-value-ml.js'  
+    import '~/components/scbd-angularjs-controls/form-control-directives/km-link-list.js'
+    import '~/views/forms/view/directives/view-record-reference.directive.js'  
+    import viewRelevantInformation from '~/views/forms/view/directives/view-relevant-information.vue';
+    import kmTerm from '~/components/km/KmTerm.vue';
+    import messages from '~/app-text/views/reports/chm/national-report-6.json';
+    import { useI18n } from 'vue-i18n';    
+    import { lstring } from '~/components/kb/filters';
+    import '~/views/forms/directives/view-terms-hierarchy.js'   
+    import KmDocumentApi from "~/api/km-document";
+    import { useAuth } from "@scbd/angular-vue/src/index.js";
+    import nationalTarget from '~/views/forms/view/chm/national-target.vue' 
+    import nationalAssessment from '~/views/forms/view/chm/national-assessment.vue' 
+
+    const auth = useAuth();
+    const kmDocumentApi = new KmDocumentApi({tokenReader:()=>auth.token()});    
+
+    const { t } = useI18n({ messages });
+
+    const props = defineProps({
+        documentInfo: { type:Object, required: true },
+        locale      : { type:String, required: true },
+    })
+    const document = computed(()=>props.documentInfo?.body);
+
+
+    const getSingleDocument= function(identifier) { 
+        return new Promise(async function (resolve, reject) {
+            const res = await kmDocumentApi.getDocument(identifier);       
+            resolve(res);
+        }) 
+    }
+
+    const nationalTargetDocumentInfos = ref([]);
+    const nationalTargetTitles = ref({});
+    const  getNationalTargetDocuments = async function() {
+        const promiseArray = document.value.nationalTargets.map(async (item) => {
+            const res = getSingleDocument(item.identifier).then((result) => {                              
+                nationalTargetDocumentInfos.value.push({body:result});
+                nationalTargetTitles.value[item.identifier]=result.title;
+            });              
+        }) 
+    }
```

Comment:
```suggestion
    const  loadNationalTargets = async function() {
        document.value.nationalTargets.map(async (item) => {
            const result = await kmDocumentApi.getDocument((item.identifier))                         
                nationalTargets.value.push({body:result});
            });              
        }) 
    }
```

why do we need a special variable for `nationalTargetTitles`?

Reply:
fixed, remove nationalTargetTitles. use nationalTargets to get the title value